### PR TITLE
Add live theme switching support

### PR DESCRIPTION
**Pull request by timja**
See https://github.com/jenkinsci/theme-manager-plugin/pull/75

All themes are now loaded on every page, the active theme is controlled by a `data` attribute on the `html` element.

FYI @janfaracik

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/darktheme/DarkThemeRootAction.java [hunk 1]
```diff
@@ -37,23 +46,27 @@ public String getUrlName() {
...
+            rsp.sendError(404);
+            return;
+        }
+        String contentType = cssFile.endsWith("map") ? "application/json" : "text/css";
+        rsp.setContentType(contentType);
+        if (Main.isDevelopmentMode) {
+            rsp.addHeader("Cache-Control", "no-cache");
+        } else {
+            rsp.addHeader("Cache-Control", "public");
         }
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/darktheme/DarkThemeRootAction.java:69]: can add in a follow up
```suggestion
```

File: src/main/java/io/jenkins/plugins/darktheme/DarkThemeRootAction.java [hunk 2]
```diff
@@ -37,23 +46,21 @@ public String getUrlName() {
...
+        if (!THEME_CSS.equals(cssFile) && !(THEME_CSS + ".map").equals(cssFile)) {
+            rsp.sendError(404);
+            return;
+        }
+        final Plugin plugin = Jenkins.get().getPlugin("dark-theme");
+        if (plugin == null) {
+            rsp.sendError(404);
+            return;
         }
+        String contentType = cssFile.endsWith("map") ? "application/json" : "text/css";
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/darktheme/DarkThemeRootAction.java:63]:
```suggestion
```

File: pom.xml [hunk 3]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.30</version>
+    <version>4.34</version>
```
Comment (timja) [inline pom.xml:7]:
```suggestion
    <version>4.37</version>
```

File: pom.xml [hunk 4]
```diff
@@ -15,20 +15,22 @@
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/dark-theme-plugin</gitHubRepo>
     <revision>0.0.3</revision>
-    <changelist>-SNAPSHOT</changelist>
-    <jenkins.version>2.303.3</jenkins.version>
+    <changelist>9999999-SNAPSHOT</changelist>
+    <jenkins.version>2.336-rc32081.db_eed62c0c0e</jenkins.version>
```
Comment (timja) [inline pom.xml:19]:
```suggestion
    <jenkins.version>2.336</jenkins.version>
```

File: pom.xml [hunk 5]
```diff
@@ -15,20 +15,22 @@
...
   </properties>
   <name>Dark Theme</name>
   <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
   <dependencyManagement>
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.303.x</artifactId>
-        <version>1008.vb9e22885c9cf</version>
+        <artifactId>bom-2.319.x</artifactId>
```
Comment (timja) [inline pom.xml:32]:
```suggestion
        <artifactId>bom-2.332.x</artifactId>
```

File: pom.xml [hunk 6]
```diff
@@ -15,20 +15,22 @@
...
+        <version>1148.v7261f385f859</version>
```
Comment (timja) [inline pom.xml:33]:
```suggestion
        <version>1155.v77b_fd92a_26fc</version>
```

File: pom.xml [hunk 7]
```diff
@@ -39,7 +41,7 @@
     <dependency>
       <groupId>io.jenkins.plugins</groupId>
       <artifactId>theme-manager</artifactId>
-      <version>0.6</version>
+      <version>0.7-rc94.26a_5398f092d</version>
```
Comment (timja) [inline pom.xml:44]:
```suggestion
      <version>1.0</version>
```

File: pom.xml [hunk 8]
```diff
@@ -15,20 +15,22 @@
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/dark-theme-plugin</gitHubRepo>
     <revision>0.0.3</revision>
```
Comment (timja) [inline pom.xml:17]:
```suggestion
```

**Comment by NotMyFault** [comment]
Looks like CD has a hard time today. Not the first release that doesn't deploy 👀

**Comment by timja** [comment]
Hmm I can't find the CD run for that one, there was a tag pushed though.

Now it's failing for expired token. I think the job doesn't run quite often enough on trusted.ci, I've triggered it now

**Comment by NotMyFault** [comment]
Interesting, I've risen https://github.com/jenkins-infra/jenkins-maven-cd-action/issues/16 earlier because of these silent fails.

**Comment by timja** [comment]
Seems the previous release did work it wasn't actually expired token. I think maybe multiple drafts broke release drafter publishing?

I can see it in the update center now

**Comment by NotMyFault** [comment]
> Seems the previous release did work it wasn't actually expired token. I think maybe multiple drafts broke release drafter publishing?

I currently have 8 drafts in https://github.com/jenkinsci/purge-build-queue-plugin/releases, if you can see that.
Kinda odd that release drafter/cd somewhat of spams drafts to then break it later on 🤔

**Comment by timja** [comment]
Yeah I can see that, (owner permission on org).

I wonder if that's breaking it, should be able to run the shell script locally to check.

**Comment by NotMyFault** [comment]
Yeah, that's it. I wiped the drafts and kept one, rebuild workflow from cli on master and it proceeded to the "release" step https://github.com/jenkinsci/purge-build-queue-plugin/runs/5325917358?check_suite_focus=true
However, that failed and created a new draft so we're back at step one 🥴

**Comment by timja** [comment]
@NotMyFault the one that failed will have actually released, you'll need to publish the release manually

**Pull request merged**